### PR TITLE
Cumsum function for DArrays

### PR DESCRIPTION
**Pull request by nschickel**
Feedback welcome, specifically related to the following:

-Flexibility of the adapted star framework for future functions
-using temporary variable tmp to transpose out3 (an array of Futures)

### CODE REVIEW

**Comment by ViralBShah** [comment]
bump @andreasnoack

File: src/core.jl [hunk 1]
```diff
@@ -711,4 +711,31 @@ Base.fill!(A::DArray, x) = begin
...
     return A
+end
+
+# new work #
+
+function star(f1,f2,f3,args::DistributedArrays.DArray...)
+        let ds = ds
+            out1 = [remotecall( ()-> f1( DistributedArrays.localpart(ds),2 )[end],p ) for p = workers()]
+
+            out2 = [0; f2(map(fetch,out1))]
```
Comment (andreasnoack) [inline src/core.jl]: The `0` prepending shouldn't be part of the `star` framework but rather part of `f2`.

File: src/core.jl [hunk 2]
```diff
@@ -711,4 +711,31 @@ Base.fill!(A::DArray, x) = begin
...
+
+            out3 = Future[]
+                    for i = 1:length(workers())
+                        p = workers()[i]
+                        out2i = out2[i]
+                        push!(out3,remotecall(()->f3(out2i,f1( DistributedArrays.localpart(ds),2) ),p))
```
Comment (andreasnoack) [inline src/core.jl]: `remotecall`s are not scheduled. You probably want to use `remotecall_wait` and a `@async` block. Look at some of the other implementations in the package to see how it works.

File: src/core.jl [hunk 3]
```diff
@@ -711,4 +711,31 @@ Base.fill!(A::DArray, x) = begin
...
+                    end
+
+            tmp = [out3[1] out3[2] out3[3] out3[4]]
```
Comment (andreasnoack) [inline src/core.jl]: Why four? And we'll need a way to generally determine the shape correctly.

**Pull request closed without merging**